logstash-core/lib/logstash/agent.rb

```diff
@@ -34,6 +36,10 @@ def execute
     start_background_services
 
     @pipelines.each { |_, p| Thread.new { p.run } }
+    @web_agent = Thread.new do
+      LogStash::WebServer::Puma.new(@logger).run
+    end
+
```

Comment:
I saw the previous note on your reviewer, so ignore previous comment I will try to reproduce your issue first.